crates/bevy_core/src/task_pool_options.rs

```diff
@@ -12,6 +13,20 @@ pub struct TaskPoolThreadAssignmentPolicy {
     /// Target using this percentage of total cores, clamped by `min_threads` and `max_threads`. It is
     /// permitted to use 1.0 to try to use all remaining threads
     pub percent: f32,
+    /// Callback that is invoked once for every created thread as it starts
+    pub on_thread_spawn: Option<Arc<dyn Fn() + Send + Sync + 'static>>,
```

Comment:
New fields are a breaking change :)

Reply:
I see. May I ask what should i put in the migration guide?

Reply:
Something to the effect of:

`TaskPooolThreadAssignmentPolicy` now has an additional field: consider defaulting it to `None`.

The main thing is to prevent us from accidentally shipping this in a minor version though.